kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/TopicEncryptionST.java

```diff
@@ -0,0 +1,122 @@
+import io.kroxylicious.systemtests.installation.kroxylicious.Kroxylicious;
+import io.kroxylicious.systemtests.installation.vault.Vault;
+import io.kroxylicious.systemtests.resources.vault.KubeVaultTestKmsFacade;
+import io.kroxylicious.systemtests.resources.vault.KubeVaultTestKmsFacadeFactory;
+import io.kroxylicious.systemtests.steps.KafkaSteps;
+import io.kroxylicious.systemtests.steps.KroxyliciousSteps;
+import io.kroxylicious.systemtests.templates.strimzi.KafkaNodePoolTemplates;
+import io.kroxylicious.systemtests.templates.strimzi.KafkaTemplates;
+import io.kroxylicious.systemtests.utils.NamespaceUtils;
+
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@ExtendWith(KroxyliciousExtension.class)
+class TopicEncryptionST extends AbstractST {
+    protected static final String BROKER_NODE_NAME = "kafka";
+    private static final Logger LOGGER = LoggerFactory.getLogger(TopicEncryptionST.class);
+    private static final String MESSAGE = "Hello-world";
+    private static KubeVaultTestKmsFacade kubeVaultTestKmsFacade;
+    private final String clusterName = "my-cluster";
+    private String bootstrap;
+
+    @BeforeAll
+    void setUp() {
+        kubeVaultTestKmsFacade = new KubeVaultTestKmsFacadeFactory().build(Vault.VAULT_DEFAULT_NAMESPACE, Vault.VAULT_POD_NAME);
+        List<Pod> vaultPods = kubeClient().listPodsByPrefixInName(Vault.VAULT_DEFAULT_NAMESPACE, Vault.VAULT_SERVICE_NAME);
+        if (!vaultPods.isEmpty()) {
+            LOGGER.warn("Skipping vault deployment. It is already deployed!");
+        }
+        else {
+            NamespaceUtils.createNamespaceWithWait(Vault.VAULT_DEFAULT_NAMESPACE);
+            kubeVaultTestKmsFacade.start();
+        }
+
+        List<Pod> kafkaPods = kubeClient().listPodsByPrefixInName(Constants.KROXY_DEFAULT_NAMESPACE, clusterName);
+        if (!kafkaPods.isEmpty()) {
+            LOGGER.warn("Skipping kafka deployment. It is already deployed!");
+            return;
+        }
+        LOGGER.info("Deploying Kafka in {} namespace", Constants.KROXY_DEFAULT_NAMESPACE);
+
+        Kafka kafka = KafkaTemplates.kafkaPersistentWithKRaftAnnotations(Constants.KROXY_DEFAULT_NAMESPACE, clusterName, 3).build();
+
+        resourceManager.createResourceWithWait(
+                KafkaNodePoolTemplates.kafkaBasedNodePoolWithDualRole(BROKER_NODE_NAME, kafka, 3).build(),
+                kafka);
+    }
+
+    @BeforeEach
+    void beforeEach(String namespace) {
+        var testKekManager = kubeVaultTestKmsFacade.getTestKekManager();
+        testKekManager.generateKek(topicName);
+
+        // start Kroxylicious
+        LOGGER.info("Given Kroxylicious in {} namespace with {} replicas", namespace, 1);
+        Kroxylicious kroxylicious = new Kroxylicious(namespace);
+        kroxylicious.deployPortPerBrokerPlainWithTopicEncryptionFilter(clusterName, 1, topicName);
```

Comment:
to obey the principle of least surprise, if beforeEach creates something, afterEach should destroy the same thing.  i.e always aim for symmetry.

So, in this case, I'd be undeploying kroxy and deleting the topic.

Reply:
This is done in `afterEach` on `KroxyliciousExtension` as we agreed on that time. We create the ephemeral namespace there and remove it at the end of each test

Reply:
okay for Kroxylicious, but the topic is still cluttter

Reply:
Done